pkg/controllers/core/deployer_controller.go

```diff
@@ -132,6 +132,7 @@ func (r *DeployerReconciler) reconcile(ctx context.Context, log logr.Logger, dep
 	childService, err := r.reconcileChildService(ctx, log, deployer)
 	if err != nil {
 		log.Error(err, "unable to reconcile child Service", "deployer", deployer)
+		deployer.Status.MarkServiceReconcileError(err)
```

Comment:
we only want to update the condition if the service resource fails to be created because the name already exists.

By always setting the condition for any error, transient errors will causes the ready condition of the deployer to turn false. The riff cli with the `--tail` flag will watch the resource until the ready condition is no longer unknown. This change as proposed will cause the CLI to report failure much more frequently.

Reply:
Makes sense. Also, that will make the condition reason `createError` instead of the rather bland `reconcileError`.

Reply:
Actually made the `ServiceReady` condition reason `serviceCreationFailed` so it makes more sense when percolated up to the `Ready` condition reason.